app/src/main/kotlin/at/bitfire/davdroid/sync/SyncNotificationManager.kt

```diff
@@ -0,0 +1,283 @@
+import dagger.hilt.android.qualifiers.ApplicationContext
+import okhttp3.HttpUrl
+import org.dmfs.tasks.contract.TaskContract
+import java.io.IOException
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class SyncNotificationManager @AssistedInject constructor(
+    @Assisted val account: Account,
+    @ApplicationContext private val context: Context,
+    private val logger: Logger,
+    private val notificationRegistry: NotificationRegistry
+) {
+
+    @AssistedFactory
+    interface Factory {
+        fun create(account: Account): SyncNotificationManager
+    }
+
+    /**
+     * Tries to inform the user that the content provider is missing or disabled.
+     */
+    fun notifyContentProviderError(authority: String) {
+        val (titleResource, textResource) = when {
+            authority == ContactsContract.AUTHORITY -> Pair(
+                R.string.sync_warning_contacts_storage_disabled_title,
+                R.string.sync_warning_contacts_storage_disabled_description
+            )
+            authority == CalendarContract.AUTHORITY -> Pair(
+                R.string.sync_warning_calendar_storage_disabled_title,
+                R.string.sync_warning_calendar_storage_disabled_description
+            )
+            TaskProvider.ProviderName.entries.map { it.authority }.contains(authority) -> Pair(
+                R.string.sync_warning_tasks_storage_disabled_title,
+                R.string.sync_warning_tasks_storage_disabled_description
+            )
+            else -> {
+                logger.log(Level.WARNING, "Content provider error for unknown authority: $authority")
+                return
+            }
+        }
+
+        notificationRegistry.notifyIfPossible(NotificationRegistry.NOTIFY_SYNC_ERROR, tag = authority) {
+            NotificationCompat.Builder(context, notificationRegistry.CHANNEL_SYNC_ERRORS)
+                .setSmallIcon(R.drawable.ic_sync_problem_notify)
+                .setContentTitle(context.getString(titleResource))
+                .setContentText(context.getString(textResource))
+                .setSubText(account.name)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setCategory(NotificationCompat.CATEGORY_ERROR)
+                .setAutoCancel(true)
+                .addAction(NotificationCompat.Action(
+                    android.R.drawable.ic_menu_view,
+                    context.getString(R.string.sync_warning_manage_apps),
+                    PendingIntent.getActivity(context, 0,
+                        Intent(Settings.ACTION_APPLICATION_SETTINGS),
+                        PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+                    )
+                ))
+                .build()
+        }
+    }
+
+    /**
+     * Tries to inform the user that an exception occurred during synchronization. Includes the affected
+     * local resource, its collection, the URL, the exception and a user message.
+     *
```

Comment:
I'd add something like:

> Use [dismissCollectionNotification] to dismiss the notification.

Same for `notifyInvalidResource`, so that it's clear to the caller what they have to do to dismiss the notification.